spec/support/extensions/esm/model/exile_territory.rb

```diff
@@ -138,23 +147,24 @@ def delete_flag
     # _flagObject setVariable ["ExileRadiusShown", false, true];
     # _flagObject setVariable ["ExileFlagStolen",_flagStolen,true];
     # _flagObject setVariable ["ExileFlagTexture",_flagTexture];
-    IGNORED_ATTRIBUTES = %w[server_id esm_custom_id]
     def update_arma
-      changed_items = previous_changes.except(*IGNORED_ATTRIBUTES)
-      return if changed_items.blank? || changed_items.key?("id")
+      changed_items = previous_changes.slice(*MAPPING.keys)
+      return if changed_items.blank?
 
       sqf = "private _flagObject = #{id} call ESMs_system_territory_get;"
-
-      if ((_, new_value) = changed_items["owner_uid"])
-        sqf += "_flagObject setVariable [\"ExileOwnerUID\", #{new_value.quoted}, true];"
-      end
-
-      if ((_, new_value) = changed_items["moderators"])
-        sqf += "_flagObject setVariable [\"ExileTerritoryModerators\", #{new_value.to_json}, true];"
-      end
-
-      if ((_, new_value) = changed_items["build_rights"])
-        sqf += "_flagObject setVariable [\"ExileTerritoryBuildRights\", #{new_value.to_json}, true];"
+      sqf += changed_items.map_join(" ") do |key, (_, value)|
+        arma_variable = MAPPING[key]
+        if arma_variable.nil?
+          warn!(
+            "ServerSetting attribute #{key.quoted} was updated but does not have a MAPPING entry"
+          )
+          next
+        end
+
+        # Ugh
```

Comment:
why ugh?

Reply:
Exile stores the value as a numeric instead of a boolean. I thought it was a boolean